docs/bookmarks.md

```diff
@@ -0,0 +1,67 @@
+# Implementation Notes:
+
+Previously only certain bookmarks were updated in Firebase:
+- personal documents loaded by the `db-other-docs-listener` bookmarks were sync'd
+- problem documents that are owned by the current user bookmarks were sync'd
+- publication document bookmarks were sync'd
+
+This was changed to update in Firebase any star the user can change in the UI.
+
+The bookmarks use Firebase as the source of truth. So when a user changes a star that star is changed in Firebase directly. The change in Firebase causes an update to the bookmarks MobX model. Firebase triggers these changes immediately even when offline so this approach should not add extra time, and it makes the code more simple.
+
+# Debugging
+
+The UI shows a bookmark on the document when any user has bookmarked the document. When a user clicks the bookmark CLUE toggles that user's bookmark on that document. If the user doesn't have a bookmark yet, then one is added. This user bookmark will be added even if the document is already showing a bookmark from a different user. All of this can be confusing.
+
+To expose this confusing logic the debug option `bookmarks` can be used. It will show a code above the bookmark on the thumbnail of the document. Like `U1 T1 O1`.
+- `U1` means the current user has a bookmark for the document
+- `T1` means 1 teacher other than you has a bookmark for the document
+- `O1` means 1 other user in the class has a bookmark for the document
+The numbers will increase if more than one teacher or other user has bookmarked the document.
+
+Additionally the `bookmarks` debug option will console log a message when the bookmark is click. This message includes all of the bookmarks on the document prior to the click.
+
+# How to represent multiple bookmarks on a single doc
+
+If we want to continue to show bookmarks made by anyone to everyone here is my proposal:
+- have 4 groups of bookmarks in the sort work tab: you, teacher, others(or other students), none
+- add some indication to the bookmark to indicate who has bookmarked it: you, teacher, others. You can see a rough demo of this with the debugging option described above.
+
+## Why you might want to bookmark something the Teacher already has:
+You want to bookmark some document even though the teacher did, because the teacher tends to removing bookmarks after that document isn't relevant anymore, and you want to keep track of it. So you still need to toggle your own bookmark of this document.
```

Comment:
```suggestion
You want to bookmark some document even though the teacher did, because the teacher tends to remove bookmarks after that document isn't relevant anymore, and you want to keep track of it. So you still need to toggle your own bookmark of this document.
```